### DISCUSSION TITLE
Hardware encoder

### DISCUSSION THREAD

**Discussion by DerekL0576** [rustdesk/rustdesk#1886] [General]
Hardware encoding should be available in the next release #589.

Is there a way I can try it out now in Windows with a nightly build?

**Comment by DerekL0576** [comment]
Through some trial and error, I managed to compile a debug build with the hardware encoder enabled.  

It was a bit of a waste of time though because the keyboard input is broken for one of the remote PCs.  For some reason I press a key and it's repeated 5 or 6 times in mapping mode and repeated twice (on key up/down?) with legacy mode.  I just tried with version 1.1.9 remote and the debug build locally and it still happens, so it's the local end that has the key repeat problem.  It's strange because it only happens with one remote PC.

The hardware encoder seems to work fine.  I'll have a look through the issues etc. to see if anybody has mentioned the input bug.

**Reply by rustdesk** [reply]
go ahead please

**Reply by djex** [reply]
I ran across this bug also. I had a brief look through the code but I didn't get far. I'm not familiar enough with rust and it's rather confusing syntax to work out what is happening. I found if I restart the rustdesk client it will "fix" the multi key presses until a keyboard mode is switched again. I will have another look to see if I can come up with anything.

**Reply by rustdesk** [reply]
Thanks, @asur4s

**Reply by chiehw** [reply]
What operating system is the remote PC

**Reply by djex** [reply]
For me it's Windows 10 Home

Edit: I have a server with several test VMs. I'll try it on Win 7, 8.1, 10 and 11 and let you know.

**Reply by chiehw** [reply]
Thx

**Reply by djex** [reply]
Ok so I tested remote on Win 7 Pro, 8.1 Pro, 10 Pro and 11 Pro. Switching between legacy and mapped keyboard modes works without producing the bug. However I did find what was causing the key repeating bug for me and I suspect it might be the same cause for @fig2k4 . 

If the remote rustdesk service crashes and restarts and the local client **auto reconnects** then any key press is doubled. If the local client is then closed, reopened and reconnected to the same remote client everything is back to normal.

To repeat the bug try:
1. Connect to remote client
2. Terminate the remote rustdesk service process. The service should auto restart.
3. Allow the local rust desk client to auto reconnect
4. All key presses should now be doubled.
5. If you repeat steps 2 and 3 again the key presses will now be tripled and so on.

I built my rustdesk using the nightly source released on Oct 20, 2022.

**Reply by chiehw** [reply]
Get it, I am solving this problem, caused by the difference between flutter and sciter
